--- ../freedom2/bootrom/freedom-u540-c000-bootloader/sifive/devices/uart.h	2019-12-19 14:29:20.358984958 +0900
+++ bootrom/freedom-u540-c000-bootloader/sifive/devices/uart.h	2019-12-18 18:48:28.280199616 +0900
@@ -14,6 +14,10 @@
 #define UART_REG_IE             0x10
 #define UART_REG_IP             0x14
 #define UART_REG_DIV            0x18
+// BA 31/10/2019 part 4
+#define UART_REG_CLKDIVSEL      0x20
+#define UART_REG_CLKDIVVAL      0x24
+// BA 31/10/2019 part 4 end
 
 /* TXCTRL register */
 #define UART_TXEN               0x1
