--- sifive-blocks/src/main/scala/devices/uart/UARTCtrlRegs.scala	2019-12-19 13:41:13.239880654 +0900
+++ ../freedom/sifive-blocks/src/main/scala/devices/uart/UARTCtrlRegs.scala	2019-12-18 18:36:25.932056122 +0900
@@ -12,4 +12,8 @@
   val ie     = 0x10
   val ip     = 0x14
   val div    = 0x18
+  //BA 31/10/2019 part 2
+  val clkdivsel = 0x20
+  val clkdivval = 0x24
+  //end BA 31/10/2019 part 2
 }
