--- sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit100MHz.dts	2019-12-27 14:11:01.604617000 +0900
+++ sifive.freedom.unleashed.DevKitU500FPGADesign_WithDevKit100MHz_change.dts	2019-12-27 14:14:46.784434936 +0900
@@ -70,7 +70,7 @@
 	};
 	L15: memory@80000000 {
 		device_type = "memory";
-		reg = <0x80000000 0x40000000>;
+		reg = <0x80000000 0x10000000>;
 	};
 	L19: soc {
 		#address-cells = <1>;
