--- bootrom/freedom-u540-c000-bootloader/sifive/platform_fpga.h	2019-12-27 13:00:12.746403956 +0900
+++ ../freedom5/bootrom/freedom-u540-c000-bootloader/sifive/platform_fpga.h	2019-12-26 17:40:21.575439028 +0900
@@ -31,7 +31,7 @@
  *****************************************************************************/
 
 // CPU info
-#define NUM_CORES 4
+#define NUM_CORES 2
 #define MAX_HART_ID 4
 #define GLOBAL_INT_SIZE 59
 #define GLOBAL_INT_MAX_PRIORITY 7
@@ -63,7 +63,7 @@
 #define CADENCEGEMGXLMGMT_CTRL_SIZE _AC(0x1000,UL)
 #define CCACHE_CTRL_ADDR _AC(0x2010000,UL)
 #define CCACHE_CTRL_SIZE _AC(0x1000,UL)*/
-#define CCACHE_SIDEBAND_ADDR _AC(0x90000000,UL)
+#define CCACHE_SIDEBAND_ADDR _AC(0x88000000,UL)
 #define CCACHE_SIDEBAND_SIZE _AC(0x1e0000,UL)
 /*#define CHIPLINK_MEM0_ADDR _AC(0x40000000,UL)
 #define CHIPLINK_MEM0_SIZE _AC(0x10000000,UL)
@@ -112,7 +112,7 @@
 #define MASKROM_MEM_ADDR _AC(0x78000000,UL)
 #define MASKROM_MEM_SIZE _AC(0x2000,UL)
 #define MEMORY_MEM_ADDR _AC(0x80000000,UL)
-#define MEMORY_MEM_SIZE _AC(0x40000000,UL)
+#define MEMORY_MEM_SIZE _AC(0x10000000,UL)
 /*#define MODESELECT_MEM_ADDR _AC(0x1000,UL)
 #define MODESELECT_MEM_SIZE _AC(0x1000,UL)
 #define MSI_CTRL_ADDR _AC(0x2020000,UL)
