--- UART.scala	2020-02-22 15:40:44.044466000 +0900
+++ UART2.scala	2020-01-03 20:37:47.822451000 +0900
@@ -65,6 +65,12 @@
 
   val div = Reg(init = UInt(divisorInit, c.divisorBits))
 
+  //BA 31/10/2019 part1
+  val clkdivselReg = Reg(init = UInt(0,3))
+  val clkdivvalReg = Reg(init = UInt(0,3))
+  val aestriggerReg =  Reg(init = UInt(0,1))
+  //end BA 31/10/2019 part 1
+
   private val stopCountBits = log2Up(c.stopBits)
   private val txCountBits = log2Floor(c.nTxEntries) + 1
   private val rxCountBits = log2Floor(c.nRxEntries) + 1
@@ -125,7 +131,13 @@
 
     UARTCtrlRegs.div -> Seq(
       RegField(c.divisorBits, div,
-                 RegFieldDesc("div","Baud rate divisor",reset=Some(divisorInit))))
+                 RegFieldDesc("div","Baud rate divisor",reset=Some(divisorInit)))),
+
+    //BA 31/10/2019 part 3
+    UARTCtrlRegs.clkdivsel -> Seq(RegField(3, clkdivselReg, RegFieldDesc("clkdivsel","Input of Clock divider select",reset=Some(0)))),
+    UARTCtrlRegs.clkdivval -> Seq(RegField(3, clkdivvalReg, RegFieldDesc("clkdivval","Input of Clock divider value",reset=Some(0)))),
+    UARTCtrlRegs.aestrigger -> Seq(RegField(1, aestriggerReg, RegFieldDesc("aestrigger","aes processing trigger signal",reset=Some(0)))),
+    //end BA 31/10/2019 part 3
   )
 }}
 
